sqlassignment3.txt

```diff
@@ -0,0 +1,353 @@
+	Extract the middle 3 characters from the string 'ABCDEFG'.
+
+declare @string varchar(7)='ABCDEFG'
+declare @lengt int =((len(@string)-2)/2)+1
+select substring(@string,@lengt,3);
+
+ 	From a table 'Employees' with a column 'FullName', write a query to extract the first name 
+
+select left(fullname,CHARINDEX(' ',fullname)) as firstname from employees
+
+
+	Extract the first 5 characters from the string 'SQL Server 2022'.
+
+declare @string varchar(15)='SQL Server 2022'
+select substring(@string,1,5);
+
+	From a 'Products' table with a 'ProductCode' column, write a query to get the first 3 characters of each product code.
+
+select SUBSTRING(ProductCode,1,3) as Productcode@3 from Products
+
+	Extract the last 4 characters from the string 'ABCDEFGHIJKL'.
+
+declare @string varchar(15)='ABCDEFGHIJKL'
+select Right(@string,4);
+
+ 	From an 'Orders' table with an 'OrderID' column (format: ORD-YYYY-NNNN), write a query to extract just the numeric portion at the end.
+
+select right(orderid,4) as NNNN from orders
+
+	Write a query to find the length of the string 'SQL Server Functions'.
+
+declare @string varchar(20)='SQL Server Functions'
+
+select len(@string);
+
+	From a 'Customers' table, find customers whose names are longer than 20 characters.
+
+select * from customers where len(CustomerName)>20
+
+ 	Compare the results of character count and byte count for the string 'SQL Server' with a trailing space.
+
+//character count do not count trailing spaces
+
+declare @string varchar(15)='SQL Server  '
+select len(@string) 
+
+//byte count, counts trailing spaces
+
+declare @string varchar(15)='SQL Server  '
+select datalength(@string) 
+
+	Write a query to find the byte count of an empty string and explain the result.
+
+declare @string varchar(15)=''
+select datalength(@string)
+
+//since there are no characters to count ,datalength is zero
+
+ 	Find the position of 'Server' in the string 'Microsoft SQL Server'
+
+declare @string varchar(25)='Microsoft SQL Server'
+declare @word varchar(15)='Server'
+select CHARINDEX(@word,@string)
+
+	From an 'Emails' table, write a query to extract the domain name from email addresses.
+
+select * from email
+select right(email,len(email)-CHARINDEX('@',email)) as domain from email 
+
+	Find the position of the first number in the string 'ABC123DEF456'.
+
+DECLARE @text VARCHAR(MAX) = 'ABC123DEF456';
+SELECT PATINDEX('%[0-9]%', @text) AS Position;
+
+	Write a query to find all product names from a 'Products' table that contain a number.
+
+select ProductName from Products where patindex('%[0-9]%',ProductName)>0
+
+	Join the strings 'SQL', 'Server', and '2022' with spaces between them.
+
+declare @string1 varchar(20)='SQL'
+declare @string2 varchar(20)='Server'
+declare @string3 varchar(20)='2022'
+declare @string4 varchar(50)=@string1+' '+@string2+' '+@string3
```

Comment:
CONCAT vs CONCAT_WS vs Simple "+" Operator ?

Reply:
CONCAT:           simply concatenates , consider null as empty string.
CONCAT_WS:    concatenates with seperator , null is ignored.
"+" Operator:    concatenates if there is no NULL values ,
                          if any , the result is NULL